modules/caddyhttp/app.go

```diff
@@ -173,7 +173,9 @@ func (app *App) Provision(ctx caddy.Context) error {
 	}
 
 	// prepare each server
+	oldContext := ctx.Context
 	for srvName, srv := range app.Servers {
+		ctx.Context = context.WithValue(oldContext, ServerCtxKey, srv)
```

Comment:
What is the purpose of this?

Reply:
In http middleware provision, this gives them to access current server by using context key. Using context value to get current server the middleware is handling request.